src/main/java/com/example/dbdemo/service/StudentServiceImpl.java

```diff
@@ -1,46 +1,73 @@
 package com.example.dbdemo.service;
 
+import com.example.dbdemo.model.Faculty;
 import com.example.dbdemo.model.Student;
 import com.example.dbdemo.repository.StudentRepository;
+import com.example.dbdemo.utilities.FilterUtils;
+import com.example.dbdemo.utilities.Gender;
 import jakarta.transaction.Transactional;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.cache.annotation.Cacheable;
+import org.springframework.data.domain.Sort;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Predicate;
 
 @Service
 public class StudentServiceImpl implements StudentService {
 
     @Autowired
     private StudentRepository studentRepository;
+    private FilterUtils<Student> filterUtil = new FilterUtils<>();
 
 
     @Override
     public List<Student> getAllStudents() {
-        return studentRepository.findAll();
+        return this.filterUtil.filterList(studentRepository.findAll(), Student::isActive);
     }
 
+    public List<Student> getAllSortedStudents(String sort){
+        return this.filterUtil.filterList(studentRepository.findAll(Sort.by(sort)), Student::isActive);
+    }
     @Override
-    @Cacheable(value = "student", key = "#id")
-//    @Transactional
+//    @Cacheable(value = "student", key = "#id")
     public Student getStudentById(Long id) {
         return studentRepository.findById(id).orElse(null);
     }
 
+    public List<Student> getFilteredStudents(String name, Gender gender) {
+        List<Student> students = studentRepository.findAll();
```

Comment:
instead of finding all the student details from the DB, can we write a separate  DB query that will return filtered students?

Reply:
Created a custom query method in StudentRepository